app/guid-file/route.ts

```diff
@@ -1,11 +1,19 @@
+import { service } from '@ember-decorators/service';
 import Route from '@ember/routing/route';
-import { inject as service } from '@ember/service';
-import Analytics from 'ember-osf-web/mixins/analytics';
+import { analyticPrivacy } from 'ember-osf-web/services/analytics';
 
-export default class FileDetail extends Route.extend(Analytics) {
-    currentUser = service('currentUser');
+export default class FileDetail extends Route.extend() {
+    @service currentUser;
+    @service analytics;
+    actions = {
+        didTransition(this: FileDetail) {
+            const publicPrivate = analyticPrivacy.public;
+            const resourceType = 'files';
```

Comment:
enum? :wink:

Reply:
I thought we decided we were going to wait on that one. That being said, if you're like an enum of jsonapi types, where would you like it to live? The analytics service isn't really the appropriate place for something that generic.